[20.8.1] Build test fails with missing initializer for field 'tcpv4soc'

The build tests fails with:
``` 
> [ 85%] Building C object boreas/CMakeFiles/util-test.dir/util_tests.c.o
> cd /builddir/build/BUILD/gvm-libs-20.8.1/boreas && /usr/lib64/ccache/cc -DGVM_LIBS_VERSION=\"20.8.1\" -DREDIS_SOCKET_PATH=\"/run/redis/redis.sock\" -D_BSD_SOURCE -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1  -m64 -mtune=generic -g -Wall -Wextra -Werror    -std=gnu99 -o CMakeFiles/util-test.dir/util_tests.c.o   -c /builddir/build/BUILD/gvm-libs-20.8.1/boreas/util_tests.c
> /builddir/build/BUILD/gvm-libs-20.8.1/boreas/util_tests.c: In function 'util__set_all_needed_sockets':
> /builddir/build/BUILD/gvm-libs-20.8.1/boreas/util_tests.c:78:10: error: missing initializer for field 'tcpv4soc' of 'struct scanner' [-Werror=missing-field-initializers]
>    struct scanner scanner = {};
>           ^
> In file included from /builddir/build/BUILD/gvm-libs-20.8.1/boreas/util.h:23:0,
>                  from /builddir/build/BUILD/gvm-libs-20.8.1/boreas/util.c:20,
>                  from /builddir/build/BUILD/gvm-libs-20.8.1/boreas/util_tests.c:20:
> /builddir/build/BUILD/gvm-libs-20.8.1/boreas/alivedetection.h:58:7: note: 'tcpv4soc' declared here
>    int tcpv4soc;
>        ^
> cc1: all warnings being treated as errors
```
OS: CentOS7

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Added
 ### Changed
 ### Fixed
+* Adding initialization to struct scanner in `boreas/util_tests.c`. [#438](https://github.com/greenbone/gvm-libs/pull/438)
+
 ### Removed
 
 [20.8.2]: https://github.com/greenbone/gvm-libs/compare/v20.8.1...gvm-libs-20.08

--- a/base/networking.c
+++ b/base/networking.c
@@ -651,7 +651,7 @@ port_range_ranges (const char *port_range)
 
       if (element_strlen >= 2)
         {
-          if ((element[0] == 'T'))
+          if (element[0] == 'T')
             {
               element++;
               while (*element && isblank (*element))
@@ -662,7 +662,7 @@ port_range_ranges (const char *port_range)
                   tcp = 1;
                 }
             }
-          else if ((element[0] == 'U'))
+          else if (element[0] == 'U')
             {
               element++;
               while (*element && isblank (*element))

--- a/boreas/util_tests.c
+++ b/boreas/util_tests.c
@@ -75,7 +75,7 @@ Ensure (util, set_all_needed_sockets)
   g_setsockopt_use_real = false;
 
   alive_test_t alive_test;
-  struct scanner scanner = {};
+  struct scanner scanner = {0};
 
   /* All methods set. */
   alive_test = ALIVE_TEST_TCP_ACK_SERVICE | ALIVE_TEST_ICMP | ALIVE_TEST_ARP

